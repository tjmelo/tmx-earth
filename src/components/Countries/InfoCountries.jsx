import { useEffect, useState } from 'react';
import { useSelector } from 'react-redux';
import axios from 'axios';

import { MountListCountries } from '../mountListCountries';
import Loading from '../Load';

export const InfoCountries = () => {

    const country = useSelector(state => state.country.country);
    const [info, setInfo] = useState([]);

    useEffect(() => {    
        // Export all details by countries...
        setInfo(<div className="alert alert-primary">Getting data...</div>)

        if(country.length === 0 || country === 'Select a country'){
            setInfo(<Loading type='warning'>Select a country!</Loading>)
        }else{
            const data = axios.get(`https://restcountries.com/v3.1/name/${country}?fullText=true`)
            data.then(({data}) => {
                setInfo(data.map((el, idx) => <MountListCountries data={el} key={idx}/> ))
            }).catch(e => console.error(e));
        }
    }, [country])

    return(
        <section className="container my-5">
            { info }
        </section>
    )
}