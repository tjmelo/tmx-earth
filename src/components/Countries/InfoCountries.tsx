import { SetStateAction, useEffect, useState } from 'react';
import { useSelector } from 'react-redux';
import axios, { AxiosResponse } from 'axios';

import MountListCountries from '../Mount';
import Loading from '../Load';

import { ICountry } from '../../interfaces';

export const InfoCountries = () => {

    const country = useSelector((state: ICountry ) => state.country.country);
    const [info, setInfo] = useState<[] | SetStateAction<any>>([]);

    const getData = async (fetchData: Promise<AxiosResponse<any, any>>) => {
        try {
            const { data } = await fetchData;
            setInfo(data?.map((el:{cca2: string}) => <MountListCountries data={el} key={el.cca2}/> ))
        } catch (e) {
            console.error(e)
        }
    }

    useEffect(() => {    
        setInfo(<div className="alert alert-primary my-2">Getting data...</div>)

        if (!country.length || country === 'Type the name of a country'){
            setInfo(<Loading type='warning'>Type the name of a country!</Loading>)
        } else {
            const data = axios.get(`https://restcountries.com/v3.1/name/${country}?fullText=true`)
            getData(data)
        }
    }, [country])

    return(
        <section className="container">
            { info }
        </section>
    )
}
