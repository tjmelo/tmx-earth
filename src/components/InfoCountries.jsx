import { useEffect, useState } from 'react';
import { useSelector } from 'react-redux';
import axios from 'axios';

import { MountListCountries } from './mountListCountries';

export const InfoCountries = props => {

    const country = useSelector(state => state.country.country);
    const [info, setInfo] = useState([]);

    useEffect(() => {    
        // Export all details by countries...
        setInfo(<div className="alert alert-primary">Getting data...</div>)

        if(country.length === 0 || country === 'Select a country'){
            setInfo(<div className="alert alert-warning">Select a country!</div>)
        }else{
            const data = axios.get(`https://restcountries.eu/rest/v2/name/${country}?fullText=true`)
            data.then(d => {
                setInfo(d.data.map((el, idx) => <MountListCountries data={el} key={idx}/> ))
            }).catch(e => console.error(e));
        }
    }, [country])

    return(
        <section className="container my-5">
            { info }
        </section>
    )
}