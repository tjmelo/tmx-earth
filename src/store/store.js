// Require configureStore from redux toolkit...
import { configureStore } from '@reduxjs/toolkit';

// Require createSlice from countryReducer...
import countryReducer from '../feature/county/countrySlice';

// Export method configureStore with object reducer...
export default configureStore({
    reducer: {
        country: countryReducer,
    }
})