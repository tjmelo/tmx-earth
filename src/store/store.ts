// Require configureStore from redux toolkit...
import { configureStore } from '@reduxjs/toolkit';

// Require createSlice from countryReducer...
import countryReducer from '../feature/country/countrySlice';

// Export method configureStore with object reducer...
export default configureStore({
    reducer: {
        country: countryReducer,
    }
})