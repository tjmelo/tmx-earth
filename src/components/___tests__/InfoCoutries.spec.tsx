import '@testing-library/jest-dom'
import { render, screen } from "@testing-library/react";
import InfoCountries from '../Countries'
import { Provider } from "react-redux";
import store from "../../store/store";

test('Should render the information of coutries', async () => {
    //do
    const {asFragment} = render(
        <Provider store={store}>
            <InfoCountries />
        </Provider>
    )

    //then    
    expect(screen.getByText('Type the name of a country!')).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot();
})
