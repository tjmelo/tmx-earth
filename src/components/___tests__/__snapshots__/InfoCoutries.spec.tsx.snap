// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the information of coutries Should render without a default value 1`] = `
<DocumentFragment>
  <section
    class="container"
  >
    <div
      class="alert alert-warning my-2"
    >
      Type the name of a country
    </div>
  </section>
</DocumentFragment>
`;
